### ISSUE TITLE
DYLD, [0x2] Wrong architecture for certain apps

### ISSUE THREAD

**Issue by bernhardsf** [JK3Y/asepsis#23]
Hi there,
Thanks again for all your help with getting asepsis running. 

I was able to successfully build and install asepsis on MacOS 10.14.5 (18F203).
Asepsis is running and the service can be restarted or stopped.

I however noticed after the install that a couple of apps on Macos stopped working. 
They are being terminated right after start. 

The crash report for those apps mention `DYLD, [0x2] Wrong architecture` 
and reference that DesktopServicesPriv that I believe are getting swapped out in the install .

```
Dyld Error Message:
  Library not loaded: /System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/DesktopServicesPriv
  Referenced from: /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/HIToolbox
  Reason: no suitable image found.  Did find:
	/System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/DesktopServicesPriv: mach-o, but wrong architecture
	/System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/DesktopServicesPriv: stat() failed with errno=1
	/System/Library/PrivateFrameworks/DesktopServicesPriv.framework/Versions/A/DesktopServicesPriv: mach-o, but wrong architecture
```

Uninstalling asepsis with `asepsisctl uninstall` and rebooting fixes the issue. 
I have attached the error log file for one of those apps. The log file is identical for all the other ones as well. 

Thanks for your help! 

file is: 
[asepsis-crash-report-steermouse.crash.txt](https://github.com/JK3Y/asepsis/files/3581424/asepsis-crash-report-steermouse.crash.txt)

**Comment by JK3Y** [comment]
Have you tried reinstalling SteerMouse after asepsis installation? It's saying the DesktopServicesPriv is wrong because that's the framework that asepsis changes.

**Comment by bernhard-adobe** [comment]
Yup, a couple of times. Is it possible that the DesktopServicesPriv needs to be a specific version for each Macos version?

**Comment by JK3Y** [comment]
no, it doesn't need to be a specific version. SteerMouse may be checking the hash of the framework and since it won't match the original (stored in ..../_A/), it's crashing. just a guess here. I thought maybe installing it after asepsis would have it think the altered desktopservicespriv was the original. I don't fully understand what the mach injection code is doing so I can't say for sure. if you uninstall SteerMouse is HIToolbox still crashing?
